assets/src/edit-story/components/canvas/layout.js

```diff
@@ -185,71 +243,148 @@ const CarouselArea = styled(Area).attrs({
-    pageSize: state.state.pageSize,
-  }));
+  const {
+    pageWidth,
+    pageHeight,
+    pagePadding,
+    viewportWidth,
+    viewportHeight,
+    hasPageNavigation,
+    hasVerticalOverflow,
+    hasHorizontalOverflow,
+    scrollLeft,
+    scrollTop,
+  } = useLayout(
+    ({
+      state: {
+        pageWidth,
+        pageHeight,
+        pagePadding,
+        viewportWidth,
+        viewportHeight,
+        hasPageNavigation,
+        hasVerticalOverflow,
+        hasHorizontalOverflow,
+        scrollLeft,
+        scrollTop,
+      },
+    }) => ({
+      pageWidth,
+      pageHeight,
+      pagePadding,
+      viewportWidth,
+      viewportHeight,
+      hasPageNavigation,
+      hasVerticalOverflow,
+      hasHorizontalOverflow,
+      scrollLeft,
+      scrollTop,
+    })
+  );
+  const fullHeight = pageWidth / FULLBLEED_RATIO;
   return {
-    '--page-width-px': `${pageSize.width}px`,
-    '--page-height-px': `${pageSize.height}px`,
-    '--fullbleed-width-px': `${pageSize.width}px`,
-    '--fullbleed-height-px': `${pageSize.width / FULLBLEED_RATIO}px`,
+    '--page-nav-width': `${hasPageNavigation ? PAGE_NAV_WIDTH : 0}px`,
+    '--page-nav-gap': `${hasPageNavigation ? PAGE_NAV_GAP : 0}px`,
+    '--page-width-px': `${pageWidth}px`,
+    '--page-height-px': `${pageHeight}px`,
+    '--page-padding-px': `${pagePadding}px`,
+    '--fullbleed-height-px': `${fullHeight}px`,
+    '--viewport-width-px': `${viewportWidth}px`,
+    '--viewport-height-px': `${viewportHeight}px`,
+    '--overflow-x': hasHorizontalOverflow ? 'scroll' : 'visible',
+    '--overflow-y': hasVerticalOverflow ? 'scroll' : 'visible',
+    '--scroll-left-px': `-${scrollLeft}px`,
+    '--scroll-top-px': `-${scrollTop}px`,
   };
 }
 
 const PageArea = forwardRef(function PageArea(
   {
     children,
-    showOverflow = false,
     fullbleedRef = createRef(),
     overlay = [],
     background,
+    isControlled = false,
+    hideScrollbars = false,
+    className = '',
+    showOverflow = false,
     isBackgroundSelected = false,
   },
   ref
 ) {
+  const { hasVerticalOverflow, hasHorizontalOverflow } = useLayout(
+    ({ state: { hasVerticalOverflow, hasHorizontalOverflow } }) => ({
+      hasVerticalOverflow,
+      hasHorizontalOverflow,
+    })
+  );
   return (
     <PageAreaFullbleedContainer
       ref={fullbleedRef}
       data-testid="fullbleed"
       aria-label={__('Fullbleed area', 'web-stories')}
-      isBackgroundSelected={isBackgroundSelected}
       role="region"
+      hideScrollbars={hideScrollbars}
+      isControlled={isControlled}
+      hasHorizontalOverflow={hasHorizontalOverflow}
+      hasVerticalOverflow={hasVerticalOverflow}
+      className={className}
     >
-      <PageAreaWithOverflow showOverflow={showOverflow} background={background}>
-        <PageAreaSafeZone ref={ref} data-testid="safezone">
-          {children}
-        </PageAreaSafeZone>
-      </PageAreaWithOverflow>
+      <PageClip
```

Comment:
Looks like something is cutting off the Media edit mode now -- it should have overflow allowed and the Zoom slider visible, currently it's hidden. Perhaps the `PageClip` is doing that?
![overflow](https://user-images.githubusercontent.com/3294597/112302677-a0c4be80-8c9b-11eb-9fc3-a52c0e24b616.gif)

Reply:
Fixed it - was tricky!